File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.py
class Solution:
    def countPartitions(self, nums: List[int]) -> int:
        res = 0
        sumR = sum(nums)
        sumL = 0
        for num in nums[:-1]:
            sumR -= num
            sumL += num
            if abs(sumL - sumR) % 2 == 0:
                res += 1
        return res

File: 3432-count-partitions-with-even-sum-difference/3432-count-partitions-with-even-sum-difference.cpp
class Solution {
public:
    int countPartitions(vector<int>& nums) {
        int res = 0;
        int sum = accumulate(nums.begin(), nums.end(), 0);
        int sumL = 0;
        int sumR = sum;
        for(int i = 0; i < nums.size() - 1; i++)
        {
            sumL += nums[i];
            sumR -= nums[i];
            if(abs(sumL - sumR) % 2 == 0)
            {
                res++;
            }
        }
        return res;
    }
};